jupyter_client/manager.py

```diff
@@ -438,16 +485,16 @@ def shutdown_kernel(self, now=False, restart=False):
         # Stop monitoring for restarting while we shutdown.
         self.stop_restarter()
 
-        self.interrupt_kernel()
+        await self._async_interrupt_kernel()
 
         if now:
-            self._kill_kernel()
+            await self._async__kill_kernel()
         else:
             self.request_shutdown(restart=restart)
             # Don't send any additional kernel kill messages immediately, to give
             # the kernel a chance to properly execute shutdown actions. Wait for at
             # most 1s, checking every 0.1s.
-            self.finish_shutdown()
+            await self._async_finish_shutdown()
```

Comment:
This breaks subclasses that override finish_shutdown() as they won't have their override called.